### PR TITLE
Add a swtich to exclude code of devtools

### PR DESCRIPTION
**Pull request by PeterWangIntel**
Build with flag '-Ddisable_devtools=1' to turn on the switch.

### CODE REVIEW

**Comment by PeterWangIntel** [comment]
@d0ngjun , thanks

File: runtime/browser/xwalk_runner.cc [hunk 1]
```diff
@@ -174,6 +176,9 @@ void XWalkRunner::OnRenderProcessHostGone(content::RenderProcessHost* host) {
 }
 
 void XWalkRunner::EnableRemoteDebugging(int port) {
+#ifdef DISABLE_DEVTOOLS
+  (void) port;
+#else
```
Comment (d0ngjun) [inline runtime/browser/xwalk_runner.cc]: remove the above two lines.

#else -> #ifndef DISABLE_DEVTOOLS.

File: xwalk.gyp [hunk 2]
```diff
@@ -319,6 +313,19 @@
         },
       },
       'conditions': [
+        ['disable_devtools!=1', {
+          'dependencies': [
+            '../components/components.gyp:devtools_http_handler',
```
Comment (d0ngjun) [inline xwalk.gyp]: why add a dependency for disable_devtools=0?

File: xwalk.gyp [hunk 3]
```diff
@@ -616,7 +623,7 @@
         'xwalk_resources',
       ],
       'conditions': [
-        [ 'OS!="android"', {
+        [ 'OS!="android"' and 'disable_devtools!=1', {
```
Comment (d0ngjun) [inline xwalk.gyp]: Seems we don't need to take this 'not android' into consideration.

File: xwalk.gyp [hunk 4]
```diff
@@ -646,7 +653,7 @@
             ],
           },
           'conditions': [
-            [ 'OS!="android"', {
+            [ 'OS!="android"' and 'disable_devtools!=1', {
```
Comment (d0ngjun) [inline xwalk.gyp]: same as above.

**Comment by d0ngjun** [comment]
Just a reminder:
Don't forget add reducing size info into git log.

File: runtime/browser/xwalk_runner.cc (hunk 1 above)
Comment (PeterWangIntel) [inline runtime/browser/xwalk_runner.cc]: yes. thanks.

File: xwalk.gyp (hunk 2 above)
Comment (PeterWangIntel) [inline xwalk.gyp]: Means, we only build it when devtools sre not enabled.

File: xwalk.gyp (hunk 3 above)
Comment (PeterWangIntel) [inline xwalk.gyp]: We need this dependency only when devtools enabled.

**Comment by PeterWangIntel** [comment]
@d0ngjun @halton

**Comment by d0ngjun** [comment]
LGTM

**Pull request merged**